res/controllers/Denon-MC7000-scripts.js

```diff
@@ -464,59 +472,27 @@ MC7000.PadButtons = function(channel, control, value, status, group) {
-        } else if (control === 0x17 && value >= 0x01) {
-            engine.setValue(group, "beatlooproll_0.5_activate", true);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x17, MC7000.padColor.rollon);
-        } else if (control === 0x17 && value >= 0x00) {
-            engine.setValue(group, "beatlooproll_0.5_activate", false);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x17, MC7000.padColor.rolloff);
-        } else if (control === 0x18 && value >= 0x01) {
-            engine.setValue(group, "beatlooproll_1_activate", true);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x18, MC7000.padColor.rollon);
-        } else if (control === 0x18 && value >= 0x00) {
-            engine.setValue(group, "beatlooproll_1_activate", false);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x18, MC7000.padColor.rolloff);
-        } else if (control === 0x19 && value >= 0x01) {
-            engine.setValue(group, "beatlooproll_2_activate", true);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x19, MC7000.padColor.rollon);
-        } else if (control === 0x19 && value >= 0x00) {
-            engine.setValue(group, "beatlooproll_2_activate", false);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x19, MC7000.padColor.rolloff);
-        } else if (control === 0x1A && value >= 0x01) {
-            engine.setValue(group, "beatlooproll_4_activate", true);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x1A, MC7000.padColor.rollon);
-        } else if (control === 0x1A && value >= 0x00) {
-            engine.setValue(group, "beatlooproll_4_activate", false);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x1A, MC7000.padColor.rolloff);
-        } else if (control === 0x1B && value >= 0x01) {
-            engine.setValue(group, "beatlooproll_8_activate", true);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x1B, MC7000.padColor.rollon);
-        } else if (control === 0x1B && value >= 0x00) {
-            engine.setValue(group, "beatlooproll_8_activate", false);
-            midi.sendShortMsg(0x94 + deckNumber - 1, 0x1B, MC7000.padColor.rolloff);
+        for (i = 0; i <= 7; i++) {
+            if (control === 0x14 + i && value > 0x00) {
+                // get initial loop size
+                // var loopSize = engine.getValue(group, "beatloop_size");
+                engine.setValue(
+                    group, "beatlooproll_" + MC7000.beatLoopRoll[i] + "_activate",
+                    true);
+                midi.sendShortMsg(0x94 + deckNumber - 1, 0x14 + i,
+                    MC7000.padColor.rollon);
+            } else if (control === 0x14 + i && value === 0x00) {
+                engine.setValue(group, "beatlooproll_activate", false);
+                midi.sendShortMsg(0x94 + deckNumber - 1, 0x14 + i,
+                    MC7000.padColor.rolloff);
+                // now set back initial loop size
+                // engine.setValue(group, "beatloop_size", loopSize);
```

Comment:
Same here

Reply:
I think it is because you are using `engine.getValue` in a loop. I'm unclear why there is a loop here at all.